test/rules/promise-async/test.ts.lint

```diff
@@ -0,0 +1,38 @@
+const nonAsyncPromiseFunction = function(p: Promise<void>) { return p; };
```

Comment:
that type is provided by `lib.es2015.d.ts`; I would prefer using that if possible instead of constructing our own type